### PR TITLE
Use assertEqual consistently in test_sparse_csr.py

### PR DESCRIPTION
**Pull request by cpuhrsch**
Let's use the provided comparison infrastructure

### CODE REVIEW

**Review by IvanYashchuk — APPROVED** [review]
Thanks!

**Review by albanD — COMMENTED** [review]
LGTM

**Review by albanD — APPROVED** [review]

File: torch/testing/_comparison.py [hunk 1]
```diff
@@ -776,11 +776,21 @@ def _compare_sparse_csr_values(
...
         - the values for closeness.
         """
+        def raise_mismatch_error(attribute_name: str, actual_value: Any, expected_value: Any) -> NoReturn:
+            raise self._make_error_meta(
+                AssertionError,
+                f"The values for attribute '{attribute_name}' do not match: {actual_value} != {expected_value}.",
+            )
+
+        if actual.shape != expected.shape:
+            raise_mismatch_error("shape", actual.shape, expected.shape)
```
Comment (pmeier) [inline torch/testing/_comparison.py:792]: @cpuhrsch This seems redundant. The comparison is dispatched from the `compare` method. In there we call `_compare_attributes` before we get to the value comparison in `_compare_sparse_csr_values`:

https://github.com/pytorch/pytorch/blob/c755d220450a8d132dff6d8a56711d93a2d17abd/torch/testing/_comparison.py#L611-L616 

`_compare_attributes` performs the exact shape you added here:

https://github.com/pytorch/pytorch/blob/c755d220450a8d132dff6d8a56711d93a2d17abd/torch/testing/_comparison.py#L641-L648

Was there a case where this was not triggered?

Comment (cpuhrsch) [reply torch/testing/_comparison.py:792]: I added this to make sure it won't be missed based on the case for the strided comparison. Happy to remove it again now that you're pointing this out.

**Pull request closed without merging**